fix(mobile): copy link prioritizes custom URLs

Lesiuk opened the pull request:
## Description

<!--- Describe your changes in detail -->
<!--- Why is this change required? What problem does it solve? -->
<!--- If it fixes an open issue, please link to the issue here. -->

The 'Copy Link' function will prioritize copying any available custom URL, mirroring the behavior found on the web version.

## How Has This Been Tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

- [x] shared_link_item_test
- [x] shared_link_edit_page_test

<details><summary><h2>Screenshots (if appropriate)</h2></summary>

<!-- Images go below this line. -->

</details>

<!-- API endpoint changes (if relevant)
## API Changes
The `/api/something` endpoint is now `/api/something-else`
-->

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)

## Please describe to which degree, if any, an LLM was used in creating this pull request.
Little help when test was failing.

bo0tzz commented on mobile/lib/widgets/shared_link/shared_link_item.dart:
```diff
@@ -77,7 +77,10 @@ class SharedLinkItem extends ConsumerWidget {
       }
 
-      Clipboard.setData(ClipboardData(text: "${serverUrl}share/${sharedLink.key}")).then((_) {
+      final linkIdentifier = sharedLink.slug ?? sharedLink.key;
+      final pathPrefix = sharedLink.slug != null ? 's/' : 'share/';
```
Having two different cases here feels brittle. Something along the lines of `if (slug) 's/' + slug else 'share/' + key` (pseudocode ofc) might be better.

Lesiuk committed:
fix: The 'Copy Link' function will prioritize copying any available custom URL, mirroring the behavior found on the web version

Lesiuk committed:
Merge branch 'main' of github.com:immich-app/immich into copy-album-url-fix

Lesiuk committed:
even simplier tests

Lesiuk committed:
refactor: organize shared link tests to follow project conventions

  - Extract mocks to shared_link_mocks.dart following existing pattern
  - Rename mock classes to ServiceMock convention
  - Move test helpers to shared utils for reusability
  - Remove explicit EasyLocalization setup, rely on TestUtils.init

bwees commented on mobile/lib/utils/url_helper.dart:
```diff
@@ -90,3 +91,13 @@ String? punycodeDecodeUrl(String? serverUrl) {
+/// Returns a path string based on the shared link's slug and key:
+/// - If slug is present: 's/${sharedLink.slug}'
+/// - Otherwise: 'share/${sharedLink.key}'
+String getShareUrlPath(SharedLink sharedLink) {
+  final slug = sharedLink.slug?.trim();
+  return slug?.isNotEmpty == true ? 's/$slug' : 'share/${sharedLink.key}';
+}
```
lets move this to the SharedLink class and have a getUrlPath() method that returns a String, similar to toString() that is there currently.

Lesiuk committed:
refactor: move getShareUrlPath to SharedLink class

Lesiuk committed:
Merge branch 'main' into copy-album-url-fix

bwees committed:
chore: whitespace cleanup

Lesiuk committed:
test: remove unnecessary documentation comments

Lesiuk committed:
Merge remote-tracking branch 'fork/copy-album-url-fix' into copy-album-url-fix
